sql/src/main/java/org/apache/druid/sql/AbstractStatement.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.sql;
+
+import org.apache.calcite.sql.parser.SqlParseException;
+import org.apache.calcite.tools.ValidationException;
+import org.apache.druid.java.util.common.logger.Logger;
+import org.apache.druid.query.QueryContext;
+import org.apache.druid.query.QueryContexts;
+import org.apache.druid.server.security.Access;
+import org.apache.druid.server.security.AuthorizationUtils;
+import org.apache.druid.server.security.ForbiddenException;
+import org.apache.druid.server.security.ResourceAction;
+import org.apache.druid.sql.calcite.planner.DruidPlanner;
+import org.apache.druid.sql.calcite.planner.PlannerContext;
+import org.apache.druid.sql.calcite.planner.PlannerResult;
+
+import java.io.Closeable;
+import java.util.Set;
+import java.util.UUID;
+import java.util.function.Function;
+
+/**
+ * Represents a SQL statement either for preparation or execution.
+ * A statement is given by a lifecycle context and the statement
+ * to execute. See derived classes for actions. Closing the statement
+ * emits logs and metrics for the statement.
+ */
+public abstract class AbstractStatement implements Closeable
+{
+  private static final Logger log = new Logger(AbstractStatement.class);
+
+  protected final SqlToolbox sqlToolbox;
+  protected final SqlQueryPlus queryPlus;
+  protected final SqlExecutionReporter reporter;
+  protected PlannerContext plannerContext;
+  protected Set<ResourceAction> queryResource;
+  protected Set<ResourceAction> resourceActions;
+
+  public AbstractStatement(
+      final SqlToolbox sqlToolbox,
+      final SqlQueryPlus sqlRequest,
+      final String remoteAddress
+  )
+  {
+    this.sqlToolbox = sqlToolbox;
+    this.queryPlus = sqlRequest;
```

Comment:
Any reason to use different variable names for the constructor param and the field?